src/ShaderCache.cpp

```diff
@@ -630,6 +630,23 @@ namespace SIE
 			auto& isShader = const_cast<RE::BSImagespaceShader&>(static_cast<const RE::BSImagespaceShader&>(shader));
 			auto* macros = reinterpret_cast<RE::BSImagespaceShader::ShaderMacro*>(defines);
 			isShader.GetShaderMacros(macros);
+			int lastIndex = 0;
+			while (macros[lastIndex].name != nullptr) {
```

Comment:
use `find_first_of`?

Reply:
For a pointer check, it's not necessarily going to be more performant I think. And we don't really need the flexibility `find_first_of` provides. Wasn't aware of find_first_of for thanks for pointing it out.